### ISSUE TITLE
Projects listing breaks after deleting a user added to a project.

### ISSUE THREAD

**Issue by rajujith** [apache/cloudstack#9974]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
UI,API
~~~

##### CLOUDSTACK VERSION

~~~
4.19.1.2
~~~

##### CONFIGURATION

Projects 

##### SUMMARY

Once you deleted a user in 'ROOT admin' role account, the projects can't be listed using UI/API. The deleted user was added to one of the projects. 

##### STEPS TO REPRODUCE

~~~
1. Add a user to the 'ROOT admin' Account.
2. Create a project as another account/user in the same domain.
3. Add the user from step 1 to this project as Project Admin.
4. As an 'admin' user delete the above user. 
~~~

##### EXPECTED RESULTS

~~~
Deleting a user shouldn't break project access. 
~~~

##### ACTUAL RESULTS

~~~
Deleting a user breaks projects access.
~~~
Undo the user deletion. ( May not be safe for production) 
Workaround example: `delete from project_account where user_id=<user_id>` OR ;`update user set state='disabled',removed=NULL where id=<id>;`

**Comment by winterhazel** [comment]
We could make the `deleteUser` API remove the user from all projects before deleting it (as we already do with accounts). To normalize environments that are affected, we could insert a query into the upgrade script. Something like:

```sql
delete from project_account where user_id in (select id from user where removed);
```

**Comment by DaanHoogland** [comment]
fixed in  #10008
